volatility3/framework/layers/intel.py

```diff
@@ -324,6 +324,39 @@ class Intel32e(Intel):
 class WindowsMixin(Intel):
     _swap_bit_offset = 32
 
+    def _get_kernel_module(self):
+        kvo = self.config.get('kernel_virtual_offset', None)
+        if kvo is None:
+            return None
+
+        for module_name in self.context.modules:
+            if self.context.modules[module_name].offset == kvo:
+                return self.context.modules[module_name]
+
+        return None
+
+    @functools.lru_cache()
```

Comment:
The `lru_cache` call should work ok, but I recall some difficulties that were encountered in certain places which is why we opt for manual storage in an instance variable in some places.  If it doesn't cause any problems then that's ok, but any weird behaviour, this would be my first culprit.

Reply:
From my understand the function result is cached per class instance so I also think it should work fine.

Reply:
The kernel variable wouldn't be used by the `Intel` class, please make it part of the `WindowsMixin`.